packages/core/test/client.spec.ts

```diff
@@ -0,0 +1,283 @@
+    projectId: projectId,
+    logger: "debug",
+  };
+  
+  const notifyClient = await NotifyClient.init(notifyParams);
+
+  const w3iClient = new Web3InboxClient(notifyClient, withDomain)
+
+  Web3InboxClient.instance = w3iClient;
+
+  await w3iClient.setAccount(withAccount);
+
+  return w3iClient;
+}
+
+export const waitForEvent = async (
+  checkForEvent: (...args: any[]) => boolean
+) => {
+  await new Promise((resolve) => {
+    const intervalId = setInterval(() => {
+      if (checkForEvent()) {
+        clearInterval(intervalId);
+        resolve({});
+      }
+    }, 50);
+  });
+};
+
+describe("Web3Inbox Core Client", () => {
+  
+
+  beforeEach(() => {
+    resetSingletonState();
+  })
+
+  describe("Init procedure", () => {
+    it("successfully inits singleton", async () => {
+      const w3iClient1 = await Web3InboxClient.init({projectId, domain: testDomain})
+      const w3iClient2 = await Web3InboxClient.init({projectId, domain: testDomain})
+
+      expect(w3iClient1).toBe(w3iClient2);
+    })
+
+    it("successfully inits a notify client", async () => {
+      const w3iClient1 = await Web3InboxClient.init({projectId, domain: testDomain})
+
+      const subs = w3iClient1.getSubscriptions();
+
+      expect(subs).toEqual([]);
+    })
+
+    it("correctly updates ready state", async () => {
+
+      expect(Web3InboxClient.clientState.isReady).toEqual(false);
+
+      // Not awaiting to be able to observe initting state.
+      Web3InboxClient.init({projectId, domain: testDomain})
+
+      expect(Web3InboxClient.clientState.initting).toEqual(true);
+
+      await waitForEvent(() => {
+	return !Web3InboxClient.clientState.initting
+      })
+
+      expect(Web3InboxClient.clientState.isReady).toEqual(true);
+    })
+
+  })
+
+  describe("Managing subscription state", () => {
+    describe("Mapping and filtering subscription state", () => {
+      // current being the domain that matches the set domain & account.
+      it("Correctly fetches current subscription", async () => {
+	const w3iClient = await initNormalInstanceW3i(testAccount, testDomain);
+	Web3InboxClient.subscriptionState.subscriptions = [testSub]
+	const sub = w3iClient.getSubscription();
+	expect(sub?.topic).toEqual(testSub.topic);
+      })
+
+      it("Correctly fetches different subscriptions", async () => {
+	const w3iClient = await initNormalInstanceW3i(testAccount, testDomain);
+	Web3InboxClient.subscriptionState.subscriptions = [testSub, testSub2]
+
+	const sub = w3iClient.getSubscription(testAccount, testDomain2);
+	expect(sub?.topic).toEqual(testTopic2);
+      })
+
+      it("Fetches all subscriptions correctly on account", async () => {
+	const w3iClient = await initNormalInstanceW3i(testAccount, testDomain);
+	Web3InboxClient.subscriptionState.subscriptions = [testSub, testSub2, testSub3]
+	const subs = w3iClient.getSubscriptions();
+
+	expect(subs.length).toEqual(2);
+
+	expect(subs.find(sub => sub.account === testSub3.account)).toBeFalsy();
```

Comment:
That makes sense, I usually use `find` because it helps with debugging. I can console.log what was found etc to see where the test was going wrong. Will update.